test/zksync-tests/Permit2Lib.test.ts

```diff
@@ -0,0 +1,583 @@
+
+describe('Permit2Lib', function () {
+
+
+    let PERMIT_TYPEHASH: string = ethers.utils.keccak256(ethers.utils.toUtf8Bytes("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)"));
+
+
+    let TOKEN_DOMAIN_SEPARATOR: string;
+    let PERMIT2_DOMAIN_SEPARATOR: string;
+    let TEST_SML_DS_DOMAIN_SEPARATOR: string;
+    let TEST_LG_DS_DOMAIN_SEPARATOR: string;
+
+    let permit2: Permit2;
+    let permit2Lib: MockPermit2LibTest;
+    let token: MockERC20;
+    let safeERC20: MockSafeERC20;
+
+    let nonPermitToken: MockNonPermitERC20;
+    let lessDSToken: MockPermitWithSmallDS;
+    let largeDSToken: MockPermitWithLargerDS;
+    let largerNonStandardDSToken: MockNonPermitNonERC20WithDS;
+
+    let amount: BigNumber = DECIMAL_MULT.mul(ethers.BigNumber.from(1000000));
+
+    let blockTimestamp: BigNumber;
+
+    before(async function () {
+
+        let permit2Address = '0x28D81506519D32a212fB098658abf4a9CCe60d59'; //precalculated
+        let deployer = new Deployer(hre, PERMIT_DEPLOYER)
+        let Permit2 = await deployer.loadArtifact("Permit2")
+        let code = await provider.getCode(permit2Address);
+        //console.log(code.length)
+        if (code.length == 2) {
+            await (await PK_OWNER.transfer({
+                to: PERMIT_DEPLOYER.address,
+                amount: ethers.utils.parseEther("1.0"),
+            })).wait();
+            permit2 = await deployer.deploy(Permit2, []) as Permit2
+            //permit2Address = permit2.address
+            // console.log("1:")
+            // console.log(permit2Address)
+            // console.log(permit2.address)
+        } else {
+            permit2 = new Contract(permit2Address, Permit2.abi, PK_OWNER) as Permit2
+            // console.log("2:")
+            // console.log(permit2Address)
+            // console.log(permit2.address)
+        }
+
+
+        permit2Lib = <MockPermit2LibTest>await deployContract("MockPermit2LibTest");
+        safeERC20 = <MockSafeERC20>await deployContract("MockSafeERC20");
+
+
+        token = <MockERC20>await deployContract("MockERC20", ["Test", "MOCK", DECIMALS]);
+
+        nonPermitToken = <MockNonPermitERC20>await deployContract("MockNonPermitERC20", ["Test", "MOCK", DECIMALS]);
+        lessDSToken = <MockPermitWithSmallDS>await deployContract("MockPermitWithSmallDS", ["Test", "MOCK", DECIMALS]);
+        largeDSToken = <MockPermitWithLargerDS>await deployContract("MockPermitWithLargerDS", ["Test", "MOCK", DECIMALS]);
+        largerNonStandardDSToken = <MockNonPermitNonERC20WithDS>await deployContract("MockNonPermitNonERC20WithDS");
+
+        TOKEN_DOMAIN_SEPARATOR = await token.DOMAIN_SEPARATOR();
+        PERMIT2_DOMAIN_SEPARATOR = await permit2.DOMAIN_SEPARATOR();
+        TEST_SML_DS_DOMAIN_SEPARATOR = await lessDSToken.DOMAIN_SEPARATOR();
+        TEST_LG_DS_DOMAIN_SEPARATOR = await largeDSToken.DOMAIN_SEPARATOR();
+
+
+        await token.connect(wallet).mint(wallet.address, amount);
```

Comment:
Does it really works rn?
```suggestion
        await (await token.connect(wallet).mint(wallet.address, amount)).wait();
```